lib/fetcher.js

```diff
@@ -0,0 +1,13 @@
+export default async function fetcher(url) {
+  const res = await fetch(url);
+
+  if (!res.ok) {
+    const error = new Error("An error occurred while fetching the data.");
+    // Attach extra info to the error object.
+    error.info = await res.json();
+    error.message = res.message;
```

Comment:
the response object og the fetch method does not have a message property. [What you are looking for is `statusText`](https://developer.mozilla.org/en-US/docs/Web/API/Response)